unstructured/partition/utils/ocr_models/ocr_interface.py

```diff
@@ -0,0 +1,120 @@
+from abc import ABC, abstractmethod
+import functools
+import importlib
+from typing import Any, List, Optional, cast
+
+from PIL import Image as PILImage
+from unstructured_inference.inference.elements import TextRegion
+from unstructured_inference.inference.layoutelement import (
+    LayoutElement,
+    partition_groups_from_regions,
+)
+
+from unstructured.documents.elements import ElementType
+
+
+class OCRAgent(ABC):
+    def __init__(self):
+        self.agent = self.load_agent()
+
+    @abstractmethod
+    def load_agent(self, language: str) -> Any:
+        pass
+
+    @abstractmethod
+    def is_text_sorted(self) -> bool:
+        pass
+
+    @abstractmethod
+    def get_text_from_image(self, image: PILImage, ocr_languages: str = "eng") -> str:
+        pass
+
+    @abstractmethod
+    def get_layout_from_image(
+        self, image: PILImage, ocr_languages: str = "eng"
+    ) -> List[TextRegion]:
+        pass
+
+    @abstractmethod
+    def get_layout_elements_from_image(
+        self, image: PILImage, ocr_languages: str = "eng"
+    ) -> List[LayoutElement]:
+        pass
+
+    @staticmethod
+    @functools.lru_cache(maxsize=None)
+    def get_instance(ocr_agent_module: str) -> "OCRAgent":
+        module_name, class_name = ocr_agent_module.rsplit(".", 1)
+        module = importlib.import_module(module_name)
```

Comment:
since this line is pretty generous in terms of code execution we should add a logic before it to validate the module name is in a whitelist. 
This whitelist could be something can be overwritten by an env so user can expand it.

Reply:
I've pushed a commit that should solve this.